Automattic/wp-calypso
# Components: Remove `Disableable` in favor of the Gutenberg's `Disabled`

#### Changes proposed in this Pull Request

A while ago, we introduced a `Disableable` component to Calypso. One would wrap one or more form controls with it, and through a single prop would be able to control whether all of them are disabled or not.

Shortly afterwards, we introduced that to Gutenberg, as an option to the already existing `Disabled` component. Since it has been available for a while, we can directly consume it instead of maintaining our own Calypso version of `Disableable`, which under the hood behaved the same way anyway. 

This PR removes `Disableable` and updates the single usage of it to use `Disabled` from Gutenberg, along with its `isDisabled` prop. For more information, see the [Disabled component docs](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/disabled#isdisabled)

#### Testing instructions

* Go to `devdocs/design/form-fields`
* Verify that the toggle examples work and look the same way as they do in production.
